src/provider/provider_file_memory.c

```diff
@@ -164,17 +163,34 @@ static umf_result_t file_initialize(void *params, void **provider) {
         goto err_free_file_provider;
     }
 
-    if (utils_set_file_size(file_provider->fd, page_size)) {
+    if (utils_set_file_size(file_provider->fd, FSDAX_PAGE_SIZE_2MB)) {
         LOG_ERR("cannot set size of the file: %s", in_params->path);
         ret = UMF_RESULT_ERROR_UNKNOWN;
         goto err_close_fd;
     }
 
-    file_provider->size_fd = page_size;
+    file_provider->size_fd = FSDAX_PAGE_SIZE_2MB;
 
     LOG_DEBUG("size of the file %s is: %zu", in_params->path,
               file_provider->size_fd);
 
+    // check if file is located on FSDAX
+    if (in_params->visibility == UMF_MEM_MAP_SYNC) {
```

Comment:
this API is broken on master - there is not MAP_SYNC - "visibility" - there is MAP_PRIVATE and MAP_SHARED. The user should not need define "SYNC" flag for FS DAX. We can detect it on our side. So if user ask for MAP_SHARED we should always map it with map_sync if available.

Reply:
Fixed.
Done.